draft-ietf-httpbis-compression-dictionary.md

```diff
@@ -293,24 +295,57 @@ For example:
 Content-Dictionary: :pZGm1Av0IEBKARczz7exkNYsZb8LzaMrV7J32a2fFG4=:
 ~~~
 
-# Negotiating the compression algorithm
+# Dictionary-Compressed Brotli
 
-When a compression dictionary is available for use for a given request, the
-algorithm to be used is negotiated through the regular mechanism for
-negotiating content encoding in HTTP.
+The "br-d" content encoding identifies a resource that is a
+"Dictionary-Compressed Brotli" stream.
 
-The dictionary to use is negotiated separately and advertised in the
-"Available-Dictionary" request header.
+A "Dictionary-Compressed Brotli" stream is a Brotli {{RFC7932}} stream for a
+response that has been compressed with an external dictionary using a
+compression window not larger than 16 MB.
+
+Clients that announce support for br-d content encoding MUST be able to
+decompress resources that were compressed with a window size of up to 16 MB.
+
+With Brotli compression, the full dictionary is available during compression
+and decompression independent of the compression window, allowing for
+delta-compression of resources larger than the compression window.
+
+# Dictionary-Compressed Zstandard
+
+The "zstd-d" content encoding identifies a resource that is a
+"Dictionary-Compressed Zstandard" stream.
 
-## Compression Algorithms
-This document introduces two new content encoding algorithms:
+A "Dictionary-Compressed Zstandard" stream is a Zstandard {{RFC8878}} stream
+for a response that has been compressed with an external dictionary using a
+compression window not larger than the larger of 8 MB and 1.25 times the size
+of the dictionary, not to exceed 128 MB.
 
-- br-d: Brotli {{RFC7932}} using an external compression dictionary and a compression window of not more than 16 MB.
-- zstd-d: Zstandard {{RFC8878}} using an external compression dictionary and a compression window of not more than 8 MB.
+Clients that announce support for zstd-d content encoding MUST be able to
+decompress resources that were compressed with a window size of up to the
+larger of 8 MB and 1.25 times the size of the dictionary, not to exceed 128 MB.
```

Comment:
```suggestion
decompress resources that were compressed with a window size of at least 8 MB
or 1.25 times the size of the dictionary, which ever is greater, up to a maximum of 128 MB.
```

Reply:
Or, with a bonus text.

```suggestion
decompress resources that were compressed with a window size of at least 8 MB
or 1.25 times the size of the dictionary, which ever is greater, up to a maximum of 128 MB.
Clients that are not willing to accept a larger window need to avoid advertising
any dictionary that would require a larger window.
```

What needs to be said about rounding?  12345 * 1.25 = what?  Are we rounding up or down, to a power of two?

Reply:
On the bonus language, I think it is a bit more subtle because they could advertise a dictionary but only `br-d` if they want to keep the window under 16 MB (and if they support both).

As far as rounding goes, as far as I understand, the window representation in the stream is where the power of two (rounding down) happens so that part of it isn't something we need to spec since accepting the larger non-rounded window is still acceptable, there is just no way to represent it in the stream.

That said, it's worth mentioning rounding on the actual 1.25 times calculation to avoid fractional bytes. I added some language to round it down (effectively what you would get with integer math on most languages).